### DISCUSSION TITLE
How to make 'one file' when having RecursionError

### DISCUSSION THREAD

**Discussion by Fradhyle** [pyinstaller/pyinstaller#8618] [Help] [answered]
I want to make single file executable with PyInstaller.
I know it could be done with `--onefile`.
But I have `RecursionError` and using this.
I know `RecursionError` can be handled with using spec file, but I don't know how to make single file executable like simply using `--onefile` parameter.
How can I do that?

**Comment by bwoodsend** [comment, marked as answer]
Uhm, to me, this translates as *How are you edit the spec file without editing the spec file?*.

If you use a minimal virtual environment instead of your large base environment, unless your dependency tree is enormous, that's normally easily enough for the recursion issue to go away.

**Reply by Fradhyle** [reply]
Oh, is that how `--onefile` work?
I thought it detects only required packages.
Thank you for the answer.

**Reply by bwoodsend** [reply]
>  I thought it detects only required packages.

That's what it tries to do but packages with optional dependencies (e.g. pandas) make that pretty much impossible.

**Reply by Fradhyle** [reply]
> > I thought it detects only required packages.
> 
> That's what it tries to do but packages with optional dependencies (e.g. pandas) make that pretty much impossible.

Oh...
That's exactly what I tried to include.
pandas and selenium
I forgot there is a bunch of dependencies of those two.
